// Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package server;

message Data {
  string date = 1;
  string uid = 2;
  string line = 3;
}

message Res {
  int32 err_code = 1;
  int32 user_num = 2;
}

message UserData {
  repeated string line_str = 1;
  int32 err_code = 2;
}

message Param {
  string name = 1;
  repeated float weight = 2;
  repeated int32 shape = 3;
}

message UserParams {
  repeated Param user_params = 1;
  int32 err_code = 2;
  string uid = 3;
}

service DataServer {
  // (Method definitions not shown)
  rpc SendData(Data) returns (Res) {}
  rpc GetUserData(Data) returns (UserData) {}
  rpc ClearUserData(Data) returns (Res) {}
  rpc GetUserParams(Data) returns (UserParams) {}
  rpc UpdateUserParams(UserParams) returns (Res) {}
}
